### PR TITLE
[ws-manager] improve tracing and fix lock

### PR DESCRIPTION
**Pull request by sagor999**
## Description

improve tracing to make it easier to debug an issue with pods stuck in terminating state

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/11787

## Release Notes

```release-note
none
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

File: components/ws-manager/pkg/manager/manager.go [hunk 1]
```diff
@@ -752,12 +752,15 @@ func (m *Manager) stopWorkspace(ctx context.Context, workspaceID string, gracePe
 
 	span, ctx := tracing.FromContext(ctx, "stopWorkspace")
 	defer tracing.FinishSpan(span, &err)
+	span.LogKV("workspaceID", workspaceID)
 
 	pod, err := m.findWorkspacePod(ctx, workspaceID)
 	if isKubernetesObjNotFoundError(err) {
+		tracing.LogError(span, err)
```
Comment (aledbf) [inline components/ws-manager/pkg/manager/manager.go:759]: [this should be logged in FinishSpan](https://github.com/gitpod-io/gitpod/blob/16cdbb02225232f156097fc7d71a3730a8a6fdf3/components/common-go/tracing/tracing.go#L90)

File: components/ws-manager/pkg/manager/manager.go [hunk 2]
```diff
@@ -752,12 +752,15 @@ func (m *Manager) stopWorkspace(ctx context.Context, workspaceID string, gracePe
...
 		return err
 	}
 	if err != nil {
+		tracing.LogError(span, err)
```
Comment (aledbf) [inline components/ws-manager/pkg/manager/manager.go:763]: idem

File: components/ws-manager/pkg/manager/manager.go (hunk 1 above)
Comment (sagor999) [reply components/ws-manager/pkg/manager/manager.go]: good catch! fixed!

File: components/ws-manager/pkg/manager/monitor.go [hunk 3]
```diff
@@ -80,8 +80,7 @@ type Monitor struct {
 	initializerMap     map[string]struct{}
 	initializerMapLock sync.Mutex
 
-	finalizerMap     map[string]context.CancelFunc
-	finalizerMapLock sync.Mutex
```
[the remark this answers is not in the document]
Comment (jenting) [reply components/ws-manager/pkg/manager/monitor.go:84]: It should be fine, we are not removing it but changing the data type to `sync.Map` https://pkg.go.dev/sync#Map

[the remark this answers is not in the document]
Comment (csweichel) [reply components/ws-manager/pkg/manager/monitor.go:84]: missed that one - sorry for the noise

**Comment by sagor999** [comment]
/hold
for clean up

**Review by aledbf — APPROVED** [review]

**Pull request merged**